### PR TITLE
Logging payments made against non-frozen baskets

### PR DESCRIPTION
**Pull request by clintonb**
We should only accept payments for frozen baskets. Accepting payments for baskets in other states leads to issues that require manual intervention. This change logs an error for these payments, and displays an error message to the user.

LEARNER-867

### CODE REVIEW

**Comment by clintonb** [comment]
FYI @edx/learner 

I want to wait a bit before merging this to dig more into this issue. If this gets merged, we ultimately leave the learner in horrible state in which they can make a payment, but not actually get the product (because we show an error!). Until someone, or the code, updates their basket status, they are screwed!

**Comment by clintonb** [comment]
Ultimately we should figure out how baskets end up paid but in a non-frozen/non-submitted state.

File: ecommerce/extensions/payment/views/cybersource.py [hunk 1]
```diff
@@ -196,6 +196,13 @@ def validate_notification(self, notification):
             if not basket:
                 logger.error('Received payment for non-existent basket [%s].', basket_id)
                 raise InvalidBasketError
+
+            if basket.status != Basket.FROZEN:
+                logger.error(
+                    'CyberSource payment received for basket [%d] which is in a non-frozen state, [%s]',
+                    basket.id, basket.status
+                )
+                raise InvalidBasketError
```
Comment (rlucioni) [inline ecommerce/extensions/payment/views/cybersource.py]: @clintonb could you merge this if this line was removed? That way we can collect some additional information without changing the system's current behavior.

**Comment by bderusha** [comment]
You basically brought up my main question here, which is what is the recourse for a learner if this error occurs?

File: ecommerce/extensions/payment/views/cybersource.py (hunk 1 above)
Comment (clintonb) [reply ecommerce/extensions/payment/views/cybersource.py]: Yeah. (okay)

**Comment by clintonb** [comment]
@rlucioni @bderusha I have removed the `raise` call. We are now just logging. Thanks for the feedback! Unless you have additional feedback, I will merge this after the build passes.

**Pull request merged**